ara/exec/parser/manifest_parser.cpp

```diff
@@ -1,23 +1,228 @@
+    const StartupConfig& other) const noexcept
 {
-    // TODO implement this method
-    return ExecutionManifest();
+    return !(*this == other);
 }
 
-MachineManifest parse_machine_manifest(const std::string& /*path*/)
+bool ExecutionManifest::Process::StartupConfig::StartupOption::operator==(
+    const StartupOption& other) const noexcept
 {
-    // TODO implement this method
+    return (kind == other.kind) && (name == other.name) && (arg == other.arg);
+}
+
+bool ExecutionManifest::Process::StartupConfig::StartupOption::operator!=(
+    const StartupOption& other) const noexcept
+{
+    return !(*this == other);
+}
+
+bool ExecutionManifest::Process::StartupConfig::MachineInstanceRef::operator==(
+    const MachineInstanceRef& other) const noexcept
+{
+    return (function_group == other.function_group) && (mode == other.mode);
+}
+
+bool ExecutionManifest::Process::StartupConfig::MachineInstanceRef::operator!=(
+    const MachineInstanceRef& other) const noexcept
+{
+    return !(*this == other);
+}
+
+namespace {
+namespace EMJsonKeys {
+const std::string kApplicationManifest = "Application_manifest";
+const std::string kApplicationManifestId = "Application_manifest_id";
+const std::string kProcess = "Process";
+const std::string kProcessName = "Process_name";
+const std::string kModeDependentStartupConfigs = "Mode_dependent_startup_configs";
+const std::string kStartupOptions = "Startup_options";
+const std::string kStartupOptionsOptionKind = "Option_kind";
+const std::string kStartupOptionsOptionName = "Option_name";
+const std::string kStartupOptionsOptionArg = "Option_arg";
+const std::string kModeInMachineInstanceRefs = "Mode_in_machine_instance_refs";
+const std::string kFunctionGroup = "Function_group";
+const std::string kMode = "Mode";
+
+const std::vector<std::string> kAsVector{kApplicationManifest,
+                                         kApplicationManifestId,
+                                         kProcess,
+                                         kProcessName,
+                                         kModeDependentStartupConfigs,
+                                         kStartupOptions,
+                                         kStartupOptionsOptionKind,
+                                         kStartupOptionsOptionName,
+                                         kStartupOptionsOptionArg,
+                                         kModeInMachineInstanceRefs,
+                                         kFunctionGroup,
+                                         kMode};
+
+}  // namespace EMJsonKeys
+
+namespace MMJsonKeys {
+const std::string kMachineManifest = "Machine_manifest";
+// TODO add rest of the keys
+
+const std::vector<std::string> kAsVector{kMachineManifest /* TODO add rest of the keys */};
+}  // namespace MMJsonKeys
+
+}  // anonymous namespace
+
+template <typename T>
+bool try_to_read_value(const json& json_obj, const std::string& key, T& output_value) noexcept
```

Comment:
Mam wrazenie ze nazwa tej funkcji odbiega od Twojej konwencji nazewnictwa. Samo read_value albo read_value_by_key czy read_mainifest_value nie pasuje lepiej i czytelniej?

Reply:
W nazwie jest `try`, bo to czytanie moze sie nie udać, jeśli klucza nie bedzie w jsonie- zależy mi, żeby to podkreślić przez taką a nie inną nazwę.
Natomiast jeśli chodzi o pozostałe propozycje, to jest to dosyc generyczna funkcja, więc nie wydaje mi się, żeby ograniczanie jej nazwą tylko do manifestu bylo potrzebne. Mozna ją wykorzystać do czytania wartosci z dowolnego jsona

Reply:
jeśli funkcja może się nie udać to polowa Twoich funkcji powinna nazywać się try_to... ;)

Reply:
Ok, w sumie może masz rację- usunę to `try_to`